### ISSUE TITLE
Rearrange e2e tests with "provisioner x source type"

### ISSUE THREAD

**Issue by akihikokuroda** [operator-framework/rukpak#580]
Every provisioner should work with any source type in theory.  Most of the source types are tested with the plain provisioner now.  So the plain_provisioner_test.go is growing.  

It's probably better that each source type is separated in a different go file and at least one test for each provisioner in it ideally.  So the xxx_provisioner_test.go can focus on the contents of the manifests instead of the source types.

**Comment by joelanford** [comment]
I agree! I think @asmacdo is working on improving the individual source unit tests (focusing on git at the moment)

I think the provisioner e2e's should focus mostly on happy path, and coverage should mainly come from unit tests when possible.

**Comment by timflannagan** [comment]
Does it make sense to enforce this for the core provisioners through a conformance testing suite?
